src/grt/src/fastroute/src/utility.cpp

```diff
@@ -253,28 +261,34 @@ void FastRouteCore::fillVIA()
             getViaStackRange(netID, node2_alias, pin_botL, pin_topL);
             bottom_layer = std::min((int) pin_botL, bottom_layer);
             top_layer = std::max((int) pin_topL, top_layer);
-          }
-          if (bottom_layer == tmpL[newCNT - 1]) {
-            bottom_layer++;
-          }
+            if (bottom_layer == tmpL[newCNT - 1]) {
```

Comment:
Is this `if` block supposed to be nested inside the `if (node2_alias < num_terminals) {`?

Reply:
yes, the only case where this can happens is if node2_alias is a terminal. It was supposed to be inside the if (node2_alias < num_terminals) before, but it wasn't and it didn't make a difference, since if node2_alias was a non terminal this statement would never enter on this if block.